### DISCUSSION TITLE
inline datums in input utxo

### DISCUSSION THREAD

**Discussion by claudiohermida** [geniusyield/atlas#305] [Q&A] [answered]
I just started using Atlas and cannot find any info on how to specify in GYTxInWitnessScript that the input to be validated has inline datum (so I shouldn't have to provide it explicitly). Would **Nothing** work?  

I'm basically looking for the equivalent of cardano-cli's "inline-datum-present"

**Comment by sourabhxyz** [comment, marked as answer]
Hi @claudiohermida, you are right that for inline datum, it is not required to be present under [`transaction_witness_set`](https://github.com/IntersectMBO/cardano-ledger/blob/e85410c8caed92b451e8657ed1465fc9a11c4930/eras/babbage/impl/cddl-files/babbage.cddl#L295-L303) and Atlas as well ignores for it [later](https://github.com/geniusyield/atlas/blob/6a4915d9312a1b9cb710ef64bc67ca28b1901e31/src/GeniusYield/Types/TxIn.hs#L82). We should thus probably modify our api to enable that possibility. I'll leave this discussion open and have created the corresponding issue [here](https://github.com/geniusyield/atlas/issues/306). Though, perhaps worth mentioning that this information is none-the-less ultimately required to correctly compute script execution units (as it is part of script context) etc. but it's not required to be explicitly given by users as we under the hood fetch details of all relevant UTxOs from their references.

**Comment by claudiohermida** [comment]
Thanks so much for your prompt response Sorabh. You are right ultimately the information needs to be queried to pass on to the execution  phase but it's a matter of streamlining a simplified UX just like cardano-cli did with their 'inline-datum-present' option. I commend the geniusyield team for providing us Haskellers with a worthy  txbuilder framework!
